### PR TITLE
Drop Rails 4.2 support

### LINKED ISSUE #202: Drop Rails 4.2 support
Rails 4.2 is now EOL.  We will be dropping support for it in v0.16.0.

If you have concerns about this, please comment on this issue.

### PR DESCRIPTION
**Pull request by olbrich**
- Drops conditional handling of Rails versions < 5.0

Fixes #202

### CODE REVIEW

File: .github/workflows/test_suite.yml [hunk 1]
```diff
@@ -2,63 +2,68 @@ name: Test Suite
...
-    branches: [ master ]
+    branches: [master]
 jobs:
   run-tests:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         orm: [active_record, mongoid]
-        rails: ['4.2', '5.0', '5.1', '5.2', '6.0']
```
Comment (olbrich) [inline .github/workflows/test_suite.yml:15]: Dropping 4.2 here and from conditionals is the major change here

File: .github/workflows/test_suite.yml [hunk 2]
```diff
@@ -2,63 +2,68 @@ name: Test Suite
...
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - uses: actions/cache@v2
+        with:
+          path: |
+            **/bundle
+          key:
+            bundle-use-ruby-${{ matrix.ruby }}-rails-${{ matrix.rails }}-orm-${{
```
Comment (oniofchaos) [inline .github/workflows/test_suite.yml:28]: Thanks for the format update here. Much easier to read. :)

File: Gemfile [hunk 3]
```diff
@@ -4,20 +4,20 @@ source 'https://rubygems.org'
...
+  gem 'sqlite3'
 end
 
 group :mongoid do
-  gem 'mongoid'
+  gem 'mongoid', '~> 6.0'
 end
 
 # This dependency is here to support an older style of testing used with Rails
 # 4.2. It can be dropped after we drop support for Rails 4.2 and we get rid of
```
Comment (oniofchaos) [inline Gemfile:19]: Can update this comment to reflect that we _just_ need to update the tests now.

File: README.md [hunk 4]
```diff
@@ -316,7 +320,7 @@ end
 
 ## Requirements
 
-- Devise (<https://github.com/plataformatec/devise>)
+- Devise (<https://github.com/heartcombo/devise>)
 - Rails 4.2 onwards (<http://github.com/rails/rails>)
```
Comment (oniofchaos) [inline README.md:324]:
```suggestion
- Rails 5 onwards (<http://github.com/rails/rails>)
```

File: test/support/integration_helpers.rb [hunk 5]
```diff
@@ -1,25 +1,18 @@
+# frozen_string_literal: true
+
 module IntegrationHelpers
   # login the user.  This will exercise all the Warden Hooks
   # @param user [User]
   # @param session [ActionDispatch::Integration::Session]
   # @return [void]
   # @note accounts for differences in the integration test API between rails versions
```
Comment (oniofchaos) [inline test/support/integration_helpers.rb:8]:
```suggestion
```

**Review by oniofchaos — COMMENTED** [review]
LOVE IT! 

One small thing I think we can clean up:

Since we aren't using Mongoid.4 anymore (since that was for Rails 4.2), we can get rid of the conditionals in these two files.

https://github.com/devise-security/devise-security/blob/f2d3fe8a4a2a28bc6764b6c3b437e3887cb4ebe0/test/dummy/config/mongoid.yml
https://github.com/devise-security/devise-security/blob/f2d3fe8a4a2a28bc6764b6c3b437e3887cb4ebe0/test/support/mongoid.yml

File: test/support/integration_helpers.rb [hunk 6]
```diff
@@ -28,20 +21,11 @@ def sign_in(user, session = integration_session)
...
```
Comment (olbrich) [inline test/support/integration_helpers.rb:22]:
```suggestion
```

**Review by oniofchaos — APPROVED** [review]
LGTM, not sure if you want to coordinate anything else before merging so I'll let you handle that.

File: Gemfile [hunk 7]
```diff
@@ -4,20 +4,21 @@ source 'https://rubygems.org'
...
+gem 'railties', '~> 5.2.4'
 
 group :active_record do
-  gem 'sqlite3', '~> 1.3.0'
+  gem 'sqlite3'
 end
 
 group :mongoid do
-  gem 'mongoid'
+  gem 'mongoid', '~> 6.0'
```
Comment (tmr08c) [inline Gemfile:15]: Curious about why we need to specify a version here

File: test/dummy/config/mongoid.yml [hunk 8]
```diff
@@ -1,5 +1,5 @@
 test:
-  <%= Mongoid::VERSION.to_i > 4 ? 'clients' : 'sessions' %>:
+  clients:
```
Comment (tmr08c) [inline test/dummy/config/mongoid.yml:2]: Is this why we specify `mongoid` to be version `6` in our base `Gemfil`?

**Review by tmr08c — COMMENTED** [review]
Late review, but 👍

**Pull request merged**